plugins/by-name/persisted/default.nix

```diff
@@ -0,0 +1,105 @@
+{
+  lib,
+  ...
+}:
+let
+  inherit (lib.nixvim) defaultNullOpts;
+  inherit (lib) types;
+in
+lib.nixvim.neovim-plugin.mkNeovimPlugin {
+  name = "persisted";
+  originalName = "persisted.nvim";
+  package = "persisted-nvim";
+
+  maintainers = [ lib.maintainers.GaetanLepage ];
+
+  settingsOptions = {
+    autostart = defaultNullOpts.mkBool true ''
+      Whether to automatically start the plugin on load.
+    '';
+
+    should_save =
+      defaultNullOpts.mkRaw
+        ''
+          function()
+            return true
+          end
+        ''
+        ''
+          Function to determine if a session should be saved
+
+          ```lua
+          @type fun(): boolean
+          ```
+        '';
+
+    save_dir =
+      defaultNullOpts.mkStr { __raw = "vim.fn.expand(vim.fn.stdpath('data') .. '/sessions/')"; }
```

Comment:
We can use the new `literalLua` for doc strings like this (https://github.com/nix-community/nixvim/pull/2520)

Reply:
Sure, done.